### DISCUSSION TITLE
Mask detection mask issue

### DISCUSSION THREAD

**Discussion by gigaklatch** [aaronwmorris/indi-allsky#1133] [Q&A]
Hello,

I am trying to get the detection mask feature working on my install of Indi-Allsky. However, as soon it switches over to night mode, I get this: 
![ccd1_20240213_180832](https://github.com/aaronwmorris/indi-allsky/assets/141551128/136c632b-df64-40c3-810f-982b32d499b6)
As opposed to how it looks when I remove the path to the mask base png:
![ccd1_20240213_180903](https://github.com/aaronwmorris/indi-allsky/assets/141551128/4ab6c816-81ec-4312-830d-3446791d0c60)

I have up until recently used a Pi HQ camera, and then I had no issues with using this feature. But now after I have done a fresh install of Indi-Allsky and am instead using a SVBony SV705C, the detection mask feature leads to the problem illustrated above.

I download the mask base file, open it in gimp, create a layer filled with black, create another layer on top of that with a white circle. And then I point Indi-Allsky to the file I export from GIMP.

Any ideas for what I might be doing wrong?

**Comment by aaronwmorris** [comment]
Can you show your mask?  A thumbnail size view is fine.

My guess is it is just the stretching.  A flat, cloudy view where there is not a lot of contrast to start with will result in this type of effect.  I would wait until you actually get some stars in the view before you make too many adjustments.

If it still bothers you, you can increase the `Stretching Std Deviations` setting in the Image tab.  Try increasing in 0.50 increments.

**Reply by gigaklatch** [reply]
Is the detection mask not only used for detecting stuff? I do not know how stuff like this works at all. But I did not think that the detection mask would be applied to the saved image, that it only would be used temporary when performing the detection.

![mask_base](https://github.com/aaronwmorris/indi-allsky/assets/141551128/c92c5d0b-0e04-49f4-a882-f42249358eda)

**Reply by aaronwmorris** [reply]
The mask is never applied to the image.  The detection mask is primarily used to limit the detection area of the image, but it is also used as the basis for the stretching operation.  Since the detection mask should only encompass "sky" this is usually a safe assumption.

The stretching operation bases the calculations on the minimum and maximum values inside the detection area, but the stretch is applied to the whole image.  In this case, since the clouds have made a very "flat" view without much contrast, the stretch tries to bring out more detail, but causes the outside edges of the image circle to be extremely dim.  When you get stars in the view where the background is darker, the effect will not be so extreme.

**Reply by gigaklatch** [reply]
Okay, I understand. 

Would it be possible to have seperate masks or areas for the detection and the measurements for the stretching?

**Reply by aaronwmorris** [reply]
It is possible, but I do not think the results would be significantly better.  You have just experienced the worst possible scenario.  Wait until you have less cloudy skies before we go down that road.

**Comment by gigaklatch** [comment]
Results from a more clear night:

Firstly, without my mask:
![DM](https://github.com/aaronwmorris/indi-allsky/assets/141551128/f31f61b0-e519-40bc-8bc8-bfdf819c28a7)

Secondly, with my mask:
![MM](https://github.com/aaronwmorris/indi-allsky/assets/141551128/568c8e96-37fd-4e12-9b75-e3f384341c76)

Quite a bit towards blue I would say.

If I again remove my mask, and then consider the SQM fov setting, which I usually have set at 66%. If I change that to anything lower than 66 % the image again suffers from that blue tint, just like when I am using my mask.  I am mentioning this finding in case it tells you anything about the situation with the mask.

**Reply by aaronwmorris** [reply]
That is interesting.  There must be something about the histogram that is causing a blue bias.

**Comment by gigaklatch** [comment]
Any suggestions for what I can do about this? You need me to provide you with anything, such as the FITS?

**Reply by aaronwmorris** [reply]
I would like to check out one of your FITS files of a clear sky.

**Reply by gigaklatch** [reply]
Github did not like me using "Attach files", so here's a link to a folder on my Google Drive:

https://drive.google.com/drive/folders/1PpXQkMRlEv0ypZcVsAxp78HL2fLSWNls?usp=sharing

Hopefully I have been able to make it accessible for you.

**Reply by aaronwmorris** [reply]
This is what I see when I debayer the image... The ZWO FITS viewer agrees with this.

The strange thing is I am pretty sure the SV705c is an IMX585, which is supposed to be RGGB, but it is detecting GRBG for the bayer pattern.

![debayer_scale](https://github.com/aaronwmorris/indi-allsky/assets/17464290/05a6269f-f42e-4372-916e-5d7226b65267)

**Reply by gigaklatch** [reply]
The SVBony SV705C does indeed have a IMX585 sensor. SVBony says on their website that the bayer pattern is GRBG [https://www.svbony.com/blog/parameters-comparison-of-sv305-to-sv705c-cameras/](url). ZWO clearly states RGGB for their ASI585MC (https://astronomy-imaging-camera.com/product/asi585mc/). Would Sony probably only be manufacturing a single version of the sensor so that all cameras with this sensor should have the very same bayer pattern?  

Could there be differences in the softwares in different cameras such that the values for the various pixels are saved in the RAW  files in different orders so that different debayering patterns for different cameras would be required, even though they have sensors that are physically the same?

One thing that may be of importance. Up until recently I was running my SV705C with a gain of 400. Looking at the graphs on the SVbony website [https://www.svbony.com/sv705c-color-planetary-camera-imx585/](url), it seems like a gain of 400 is quite high and has very low dynamic range, around 9. I now have lowered the gain to 275 which supposedly is when the HCG mode kicks in and the dynamic range is at about 11. Could this perhaps have an effect on how the stretching of the images? Since I had the gain lowered there have been only clouds, therefore I have not had the chance to see if any effects it would have on images from a clear night.

**Reply by aaronwmorris** [reply]
As for the bayer pattern... ZWO or SVBony is probably using a different offset on the sensor.  Below is RGGB with an 0,0 offset.  If you offset X+1, then the pattern is indeed GRBG.  (The pattern is 2 on the first row, 2 on the second row).  You can achieve all 4 bayer patterns by just using different offsets (or reading the sensor backwards).

```
RGRG
GBGB
RGRG
GBGB
```

I just don't know why my colors differ from yours... I will ponder this.

**Reply by gigaklatch** [reply]
Alright, got ya. That's cool.

I think the difference is automatic white balance. Try using that.
